### ISSUE TITLE
Fails in child table

### ISSUE THREAD

**Issue by JRII972** [frappe/print_designer#72]
Good morning
I can't configure the application properly.

This follows a request from a month ago, which I have not been able to reopen. [https://github.com/frappe/print_designer/issues/64](url)
When I select a field in a child table I get this error

```javascript
Uncaught (in promise) TypeError: Cannot read properties of undefined (reading 'is_primary_phone')
    at Proxy.selectField (<anonymous>:48677:246)
    at onClick (<anonymous>:48907:57)
    at callWithErrorHandling (<anonymous>:27015:24)
    at Object.callWithAsyncErrorHandling (<anonymous>:27023:23)
    at HTMLDivElement.invoker (<anonymous>:34753:23)
``` 

I found that the source of the error would be a variable ```_a2``` which would be a dictionary, empty and therefore we cannot access the value, I cannot find where it should be assigned .

```javascript
...
const selectField = async (field, fieldtype) => {
        var _a2, _b, _c, _d;
        let isRemoved = false;
        fieldnames.value = fieldnames.value.filter((el) => {
          if (el.parentField == previewRef.value.parentField && el.fieldname == field.fieldname) {
            isRemoved = true;
            MainStore.dynamicData.splice(MainStore.dynamicData.indexOf(el), 1);
            if (previewRef.value.selectedEl && el == previewRef.value.selectedEl.field) {
              previewRef.value.setSelectedEl(null);
            }
            return false;
          }
          return true;
        });
        if (isRemoved)
          return;
        let index = fieldnames.value.length;
        let value = previewRef.value.parentField ? await getValue(doctype.value, MainStore.docData[previewRef.value.parentField], field.fieldname) : props.table ? typeof ((_a2 = MainStore.docData[props.table.fieldname]) == null ? void 0 : _a2[0][field.fieldname]) != "undefined" ? frappe.format(
          MainStore.docData[props.table.fieldname][0][field.fieldname],
          { fieldtype: field.fieldtype, options: field.options },
          { inline: true },
          MainStore.docData
        ) : `{{ ${field.fieldname} }}` : frappe.format(
          MainStore.docData[field.fieldname],
          { fieldtype: field.fieldtype, options: field.options },
          { inline: true },
          MainStore.docData
        );
...
``` 
 ```_a2[0][field.fieldname]``` here that poses a problem

**Comment by JRII972** [comment]
So i found out, if the child table of the selected enregistrement is empty, the come out...
So i have to fill up all child table before

**Comment by maharshivpatel** [comment]
I understand the issue now and i am able to replicate this as well, i will fix this soon. meanwhile, use document that have at-least one row added. example document that have at-least 1 phone number added.

**Comment by JRII972** [comment]
yep, this And sorry for the trouble

**Comment by JRII972** [comment]
so so so, i was trying to close the modal and ... 

```javascript
Uncaught TypeError: Cannot read properties of undefined (reading 'indexOf')
    at eval (eval at <anonymous> (eee:1:1), <anonymous>:1:39)
    at <anonymous>:48791:11
    at Proxy.forEach (<anonymous>)
    at cancelClick (<anonymous>:48790:26)
    at callWithErrorHandling (<anonymous>:27015:24)
    at callWithAsyncErrorHandling (<anonymous>:27023:23)
    at emit (<anonymous>:27506:11)
    at <anonymous>:34037:51
    at <anonymous>:47782:42
```

**Comment by maharshivpatel** [comment]
@JRII972 this is with child table row or without ?

**Comment by JRII972** [comment]
i "fix" the previous error with a try and catch, and try both situation.

It fail when you close the modale, or cancel, after selecting content.
```javascript
let value = null;
	if (previewRef.value.parentField){
		value = await getValue(doctype.value, MainStore.docData[previewRef.value.parentField], field.fieldname)
	} else if (props.table) {
		try{
			if (typeof MainStore.docData[props.table.fieldname]?.[0][field.fieldname] != undefined ) {
			value = frappe.format(
					MainStore.docData[props.table.fieldname][0][field.fieldname],
					{ fieldtype: field.fieldtype, options: field.options },
					{ inline: true },
					MainStore.docData
				)
			}else {
				value =  `{{ ${field.fieldname} }}`
			}
		}catch{
			value =  `{{ ${field.fieldname} }}`
		}
			
	}else {
		value = frappe.format(
							MainStore.docData[field.fieldname], 	
							{ fieldtype: field.fieldtype, options: field.options },
							{ inline: true },
							MainStore.docData
					  );
	}
``` 

So i "fix" cancelClick with another try and catch
```javascript
const cancelClick = () => {
	props.openDynamicModal.dynamicContent?.forEach((element) => {
		if (MainStore.dynamicData.indexOf(element) == -1) {
			MainStore.dynamicData.push(element);
		}
	});
	fieldnames.value.forEach((element) => {
		try {
			if (props.openDynamicModal.dynamicContent.indexOf(element) == -1) {
				MainStore.dynamicData.splice(MainStore.dynamicData.indexOf(element), 1);
			}
		} catch (error) {
			console.log(error)
		}
	});
	props.openDynamicModal.selectedDynamicText = null;
	MainStore.openDynamicModal = null;
	MainStore.openTableColumnModal = null;
};
```

**Comment by maharshivpatel** [comment]
don't try and catch just check if row is available and set the value else set field_name similar to this : https://github.com/frappe/print_designer/blob/02c8eadd42e980a675204b647d2658bcb8dbce70/print_designer/public/js/print_designer/components/layout/AppBarcodeModal.vue#L194C33-L194C33

**Comment by maharshivpatel** [comment]
Fixed by #75
